packages/oui-modal/_mixins.less

```diff
@@ -107,14 +107,37 @@
     box-shadow: 0 5px 15px rgba(0, 0, 0, 0.5);
   }
 
-  .modal-warning() {
-    &__body > .oui-icon-warning_circle {
+  .modal-icon-base(
+    @icon-selector ,
+    @icon-color
+  ) {
+    &__body > .@{icon-selector} {
       position: absolute;
       display: block;
       top: @oui-modal-icon-position-top;
       left: @oui-modal-icon-position-left;
-      color: @oui-modal-icon-warning-color;
+      color: @icon-color;
       font-size: @oui-modal-icon-font-size;
     }
   }
+
+  .modal-success {
```

Comment:
instead of having those 5 mixins I would have added this

```less
#oui > .modal-base(@oui-button-selector);
#oui > .modal-icon-base(oui-icon-success_circle, @oui-modal-icon-success-color);
#oui > .modal-icon-base(oui-icon-error_circle, @oui-modal-icon-error-color);
#oui > .modal-icon-base(oui-icon-warning_circle, @oui-modal-icon-warning-color);
#oui > .modal-icon-base(oui-icon-info_circle, @oui-modal-icon-info-color);
#oui > .modal-icon-base(oui-icon-help_circle, @oui-modal-icon-help-color);
```

to *modal.less* files and  it  would keep selectors outside of mixins.

Reply:
Done